core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -241,13 +241,22 @@ void killByKiller() throws InterruptedException {
          * null if no one objects killing the process.
          */
         protected @CheckForNull VetoCause getVeto() {
-            for (ProcessKillingVeto vetoExtension : ProcessKillingVeto.all()) {
-                VetoCause cause = vetoExtension.vetoProcessKilling(this);
-                if (cause != null) {
-                    if (LOGGER.isLoggable(FINEST))
-                        LOGGER.finest("Killing of pid " + getPid() + " vetoed by " + vetoExtension.getClass().getName() + ": " + cause.getMessage());
-                    return cause;
+            String causeMessage = null;
+            
+            try {
+                VirtualChannel channelToMaster = SlaveComputer.getChannelToMaster();
+                if (channelToMaster!=null) {
+                    CheckVetoes vetoCheck = new CheckVetoes(this);
+                    causeMessage = channelToMaster.call(vetoCheck);
```

Comment:
1 minor bit here. Would it be possible to avoid calls to master when there is no `ProcessKillingVeto` implementations? Just by passing a flag somehow. Otherwise every process termination will be doing a call to the master.

Reply:
I've been meaning to look at this, not sure off the top of my head where that would be done and if it would require an extra parameter somewhere. I'll have to see which class is actually created from the master since the flag would need to be added there.  I have no idea how to test that the call doesn't occur though since the end result would be the same.